src/org/omegat/core/data/PluginInformation.java

```diff
@@ -166,11 +235,59 @@ public boolean equals(Object obj) {
         } else if (!name.equals(other.name))
             return false;
         if (version == null) {
-            if (other.version != null)
-                return false;
-        } else if (!version.equals(other.version))
-            return false;
-        return true;
+            return other.version == null;
+        } else return version.equals(other.version);
     }
 
+    @Override
+    public final int compareTo(PluginInformation pluginInformation) {
```

Comment:
I think this method can be simplified greatly by use of [Comparator](https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html).